src/IO/ReadWriteBufferFromHTTP.h

```diff
@@ -137,6 +138,7 @@ namespace detail
                 istr = receiveResponse(*sess, request, response, true);
                 response.getCookies(cookies);
 
+                compress_method = response.get("Content-Encoding");
```

Comment:
Maybe better to name it contentEncoding, because we get entity header with "Content-Encoding" name, and for client it will be easier to understand.